src/StoreKeeper/WooCommerce/B2C/Frontend/assets/storekeeper-address-form.js

```diff
@@ -0,0 +1,135 @@
+jQuery(function ($) {
+    /* Global functions */
+    window.storekeeperTranslations = settings.translations;
+    if (typeof storekeeperTranslate != 'function') {
+        window.storekeeperTranslate = function (key) {
+            if (typeof window.storekeeperTranslations[key] === 'undefined') {
+                console.log(`Storekeeper: Missing translation for (${key})`);
+            }
+            return window.storekeeperTranslations[key] ?? key;
+        }
+    };
+
+    if (typeof storekeeperBlockForm != 'function') {
+        window.storekeeperBlockForm = function (addressForm) {
+            $(':focus').blur();
+            addressForm.addClass('storekeeper-loading');
+        };
+    }
+
+    if (typeof storekeeperUnblockForm != 'function') {
+        window.storekeeperUnblockForm = function (addressForm) {
+            addressForm.removeClass('storekeeper-loading');
+        };
+    }
+
+    if (typeof storekeeperFetchAddressFromBackend != 'function') {
+        window.storekeeperFetchAddressFromBackend = function (formPrefix, parentForm) {
+            return new Promise(function (resolve, reject) {
+                let isValid = true;
+                window.storekeeperBlockForm(parentForm);
+                const postCode = $(`#${formPrefix}_postcode`).val();
+                const houseNumber = $(`#${formPrefix}_address_house_number`).val();
+                $(`#${formPrefix}_postcode`).parent().parent().removeClass('woocommerce-validated').removeClass('woocommerce-invalid');
+                $(`#${formPrefix}_address_house_number`).parent().parent().removeClass('woocommerce-validated').removeClass('woocommerce-invalid');
+                $(`#${formPrefix}_address_house_number`).parent().find('.postcode-housenr-validation-message').remove();
+                $(`#${formPrefix}_address_house_number`).parent().append(`<small class="postcode-housenr-validation-message">${window.storekeeperTranslate('Validating postcode and house number. Please wait...')}</small>`);
```

Comment:
will this work for any theme? looks really bound to the html struture

Reply:
Changed getting of container to `{field}_field` as it is woocommerce' element ID. So no longer bound to HTML structure.